### PR TITLE
faultInjectionFix-ruleMismatchOnAddressUri

### PR DESCRIPTION
**Pull request by xinlian12**
**Issue:**
In recent test, I find out sometimes the rule is not correctly applied when comparing the addressUri. 

**Root cause**:
for addressUri match, currently we are using `requestcontext.storePhysicalAddressUri`. But it can be override by different network request as they share the same RxDocumentServiceRequest.

Solution:
Using `RntbdRequestArgs.storePhysicalAddressUri` instead

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-test/src/test/java/com/azure/cosmos/test/faultinjection/FaultInjectionServerErrorRuleTests.java [hunk 1]
```diff
@@ -66,7 +66,7 @@ public FaultInjectionServerErrorRuleTests(CosmosClientBuilder clientBuilder) {
         this.subscriberValidationTimeout = TIMEOUT;
     }
 
-    @BeforeClass(groups = {"multi-region"}, timeOut = TIMEOUT)
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-test/src/test/java/com/azure/cosmos/test/faultinjection/FaultInjectionServerErrorRuleTests.java:69]: why this change in this test class?

File: sdk/cosmos/azure-cosmos-test/src/test/java/com/azure/cosmos/test/faultinjection/FaultInjectionServerErrorRuleTests.java [hunk 2]
```diff
@@ -620,23 +621,24 @@ public void faultInjectionServerErrorRuleTests_ServerConnectionDelay_warmup(
...
 
-            CosmosFaultInjectionHelper.configureFaultInjectionRules(container, Arrays.asList(serverConnectionDelayRule)).block();
+            CosmosFaultInjectionHelper.configureFaultInjectionRules(container, Arrays.asList(serverConnectionDelayWarmupRule)).block();
 
             int partitionSize = container.getFeedRanges().block().size();
             container.openConnectionsAndInitCaches().block();
 
             if (primaryAddressesOnly) {
-                assertThat(serverConnectionDelayRule.getHitCount()).isEqualTo(partitionSize);
+                System.out.println("Partition size is " + partitionSize);
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-test/src/test/java/com/azure/cosmos/test/faultinjection/FaultInjectionServerErrorRuleTests.java:630]: Remove the system log.

**Review by FabianMeiswinkel — APPROVED** [review]
Thx

File: sdk/cosmos/azure-cosmos-test/src/test/java/com/azure/cosmos/test/faultinjection/FaultInjectionServerErrorRuleTests.java (hunk 1 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-test/src/test/java/com/azure/cosmos/test/faultinjection/FaultInjectionServerErrorRuleTests.java:69]: I want to test multi-region and single region, that is why i changed to have both

File: sdk/cosmos/azure-cosmos-test/src/test/java/com/azure/cosmos/test/faultinjection/FaultInjectionServerErrorRuleTests.java (hunk 2 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-test/src/test/java/com/azure/cosmos/test/faultinjection/FaultInjectionServerErrorRuleTests.java:630]: Will remove in the next PR
